### PR TITLE
Added contextual bandit testbed

### PR DESCRIPTION
**Pull request by threewisemonkeys-as**
1. Added a simple testbed for both `GaussianCB` and `BernoulliCB` which just give a random bandit as context every timestep. 
2. Added simple eps greedy and UCB policies which keep track of the value approximations in a table
3. Added a generic `BaseValue` class in `values.py` <-- Not quite sure how useful this will be 

I was going through [this](https://github.com/tensorflow/models/tree/master/research/deep_contextual_bandits) deep CB paper and code for reference to see how we can incorporate deep CB into this structure.

### CODE REVIEW

**Comment by threewisemonkeys-as** [comment]
> Looks pretty good to me. Can you upload the plots as well?

Should I add these in a `/bandit/results` folder?

File: genrl/classical/bandit/contextual_policies.py [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+        self.reward_hist.append(reward)
+        self._regret += max(self.Q[context]) - self.Q[context, action]
+        self.regret_hist.append(self.regret)
+        self.Q[context, action] += (reward - self.Q[context, action]) / (
+            self.counts[context, action] + 1
+        )
+        self.counts[context, action] += 1
+
+
+if __name__ == "__main__":
```
Comment (Sharad24) [inline genrl/classical/bandit/contextual_policies.py:297]: Everything after this can be shifted to a separate file in the directory `examples`.

**Comment by Sharad24** [comment]
> Should I add these in a `/bandit/results` folder?

No, I was just asking this for us to see the results. You can upload it here in the chat itself

**Comment by threewisemonkeys-as** [comment]
EpsGreedyCBPolicy-on-BernoulliCB
EpsGreedyCBPolicy-on-GaussianCB
UCBCBPolicy-on-BernoulliCB
UCBCBPolicy-on-GaussianCB

**Comment by threewisemonkeys-as** [comment]
@Sharad24  Should I refactor `contextual_policies.py` into multiple files to resolve the codeclimate issue?

**Comment by Sharad24** [comment]
> @Sharad24 Should I refactor `contextual_policies.py` into multiple files to resolve the codeclimate issue?

Yeah I think that should be good. I'm going to merge #168 which fixes an existing code climate issue. What you can do after that is add each policy to its respective file inside `policies`. If files exceed 250 lines in that, then I think we should make another folder called `contextual_policies` and follow a structure similar to `policies` in #168

**Comment by Sharad24** [comment]
Apologies for not asking you to separate `policies`, I didnt know the errors here were coming from a similar issue

**Comment by threewisemonkeys-as** [comment]
Pretty much done now. The only issue is coming from `examples/bandit.py` which has a somewhat big function to run and lot policies. I dont really see any way of simplifying apart from breaking it up which would be a little cumbersome. Is it really necessary to refactor it?

**Comment by Sharad24** [comment]
Its probably ok for now. Among other things, its a very small issue. Great work! @threewisemonkeys-as

**Pull request merged**